### PR TITLE
[MS-353] Deploy Moonshot to EKS

### PR DESCRIPTION
**Pull request by imda-lionelteo**
## Description

Deploy Moonshot to EKS

## Motivation and Context

This is to prepare moonshot to deploy to CStack or other ways that external parties want to deploy as an application in their own cluster.

This is an example of how they can deploy moonshot in their kubernates cluster.

## Type of Change

- feat: A new feature

## How to Test

Deploy the kubernates scripts
Load the website url and test it out.

## Checklist

Please check all the boxes that apply to this pull request using "x":

- [x]  I have tested the changes locally and verified that they work as expected.
- [ ]  I have added or updated the necessary documentation (README, API docs, etc.).
- [ ]  I have added appropriate unit tests or functional tests for the changes made.
- [x]  I have followed the project's coding conventions and style guidelines.
- [ ]  I have rebased my branch onto the latest commit of the main branch.
- [ ]  I have squashed or reorganized my commits into logical units.
- [ ]  I have added any necessary dependencies or packages to the project's build configuration.
- [x]  I have performed a self-review of my own code.
- [x]  I have read, understood and agree to the Developer Certificate of Origin below, which this project utilises.

## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>

### CODE REVIEW

File: examples/deployment/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Use an official Python runtime as a parent image
+FROM python:3.11-slim AS build
+
+# patches & OS dependencies
+RUN apt-get update && apt-get -y upgrade && \
+    apt-get install -y git build-essential pkg-config libhdf5-dev npm && \
+    rm -rf /var/lib/apt/lists/*
+
+# python dependencies
+RUN pip3 install --no-cache-dir --upgrade pip && \
```
Comment (imda-kwokwk) [inline examples/deployment/Dockerfile:10]: Do you need to use venv?

Comment (imda-lionelteo) [reply examples/deployment/Dockerfile:10]: hmm i consider as the requirements are all in the container and isolated, there is no need to actually isolate them figure. as for our own local machine, we will need to isolate requirements from different build.

i would actually defer to your opinion on whether need to venv or not but that is my reasoning

Comment (imda-kwokwk) [reply examples/deployment/Dockerfile:10]: thanks, looks good then.

**Review by imda-kwokwk — APPROVED** [review]
looks good

**Comment by imda-benedictlee** [comment]
Link to Jira Ticket: https://imda-dsl.atlassian.net/browse/MS-353?atlOrigin=eyJpIjoiZTg5ZWJhYzI2NjQxNDc0MGEwODdhNDk3MjYwZWJhOTMiLCJwIjoiaiJ9

**Comment by imda-benedictlee** [comment]
Tested. Working as expected. Screen recording at the above Jira Ticket.

**Pull request merged**